### PR TITLE
Fix test error - Issue hiker#4, hiker#5

### PR DESCRIPTION
**Pull request by jasonjunweilyu**
This PR fix all the test errors. The detailed error messages are listed in the comment section of each commit and also shown in Issue hiker#4. Now with `pytest`, the tests can be run with `pytest tests` for all the tests and `pytest tests/unit_tests` or `pytest tests/system_tests` for a specific test group.

### CODE REVIEW

File: tests/system_tests/psyclone/test_psyclone_system_test.py [hunk 1]
```diff
@@ -160,17 +161,17 @@ def test_run(self, config):
...
-            config.prebuild_folder / 'algorithm_mod_psy.5088673431.f90',  # prebuild
+            # The kernel hash differs between fpp and cpp, so just use wildcards.
+            config.prebuild_folder / 'algorithm_mod.*.an',  # x90 analysis result
+            config.prebuild_folder / 'my_kernel_mod.*.an',  # kernel analysis results
+            config.prebuild_folder / 'algorithm_mod.*.f90',  # prebuild
+            config.prebuild_folder / 'algorithm_mod_psy.*.f90',  # prebuild
         ]
 
-        assert all(not f.exists() for f in expect_files)
+        assert all(not any(glob.glob(str(f))) for f in expect_files)
```
Comment (hiker) [inline tests/system_tests/psyclone/test_psyclone_system_test.py:171]: Is this `any` actually required? glob seems to return a list (not an iteratable), and as far as I can see `not glob...` should be the same result as `not any(glob...)`? Or am I missing something?

File: tests/system_tests/psyclone/test_psyclone_system_test.py [hunk 2]
```diff
@@ -160,17 +161,17 @@ def test_run(self, config):
...
         with config:
             self.steps(config)
-        assert all(f.exists() for f in expect_files)
+        assert all(any(glob.glob(str(f))) for f in expect_files)
```
Comment (hiker) [inline tests/system_tests/psyclone/test_psyclone_system_test.py:174]: Similar question here - while the `any` is required to convert a list to true/false, I think it makes it harder to understand what this line does. Can this be replaced with:
```
all(glob.glob(str(f)) != []) for f in expect_files)
```

File: tests/system_tests/svn_fcm/test_svn_fcm_system_test.py
Comment (hiker) [inline tests/system_tests/svn_fcm/test_svn_fcm_system_test.py:1]: Great to remove this duplication, thanks!

**Review by hiker — CHANGES_REQUESTED** [review]
Hi,
thanks for this, The patch itself works, i.e. I don't get failure with psyclone 2.5 anymore. I suggested two code simplifications ... I think :) Let me know what you think.

File: tests/system_tests/psyclone/test_psyclone_system_test.py (hunk 2 above)
Comment (jasonjunweilyu) [reply tests/system_tests/psyclone/test_psyclone_system_test.py]: Thanks for the great suggestion. The original fix is indeed hard to understand. When I now look at my fix, I also spent some time trying to see if there is any reason for adding the `any`. I have updated this with a new commit.

File: tests/system_tests/psyclone/test_psyclone_system_test.py (hunk 1 above)
Comment (jasonjunweilyu) [reply tests/system_tests/psyclone/test_psyclone_system_test.py]: Thanks! I have improved this according to the comment below in a new commit. Now it is ready for you to review.

File: tests/system_tests/psyclone/test_psyclone_system_test.py [hunk 3]
```diff
@@ -160,17 +161,17 @@ def test_run(self, config):
...
-            config.prebuild_folder / 'algorithm_mod_psy.5088673431.f90',  # prebuild
+            # The kernel hash differs between fpp and cpp, so just use wildcards.
+            config.prebuild_folder / 'algorithm_mod.*.an',  # x90 analysis result
+            config.prebuild_folder / 'my_kernel_mod.*.an',  # kernel analysis results
+            config.prebuild_folder / 'algorithm_mod.*.f90',  # prebuild
+            config.prebuild_folder / 'algorithm_mod_psy.*.f90',  # prebuild
         ]
 
-        assert all(not f.exists() for f in expect_files)
+        assert all((glob.glob(str(f)) == []) for f in expect_files)
```
Comment (hiker) [inline tests/system_tests/psyclone/test_psyclone_system_test.py:171]: That's even better, thanks :)

**Review by hiker — APPROVED** [review]
All good, proceeding to merge

**Pull request merged**